### PR TITLE
`zea.ops` subpackage and beamform pipeline refactor

### PR DESCRIPTION
**Pull request by tristan-deep**
Related to #145 and #143. Reason they are not in two separate PRs is because refactoring the beamforming resulted in lots of circular imports which are all resolved with the redesign of the `zea.ops` subpackage.

## Changes
1. Addressed #145. Cleaned up all ops modules and organized into two subpackages:
- `zea.ops`, which are all`Operation`(s) which should be used in a `Pipeline`. Organized into:
    - ultrasound operations (`zea.ops.ultrasound`, but just exposed as `zea.ops`)
    - tensor operations (`zea.ops.tensor`, but just exposed as `zea.ops`)
    - base operations (`zea.ops.base`, but just exposed as `zea.ops`)
    - keras operations, autogenerated (`zea.ops.keras_ops`, but just exposed as `zea.ops`)
- `zea.func`, which are all functional operations and just standalone funcs.
    - ultrasound specific (`zea.func.ultrasound`, but just exposed as `zea.func`)
    - tensor operations (`zea.func.tensor`, but just exposed as `zea.func`, this used to be `zea.tensor_ops`)
- Examples:
    - `zea.ops.Demodulate` and `zea.func.demodulate`, which live in `zea/ops/ultrasound.py` and `zea/func/ultrasound.py`.
    - `zea.ops.Sum` and `zea.func.sum`, which live in `zea/ops/tensor.py` and `zea/func/tensor.py`.

2. Addressed #143 
- You can now import: `from zea.ops import Beamform`, which is a pipeline as follows (actually with options):

```python
from ops import Beamform
beamforming = Beamform(
    beamformer="das",
    enable_pfield=False,
    num_patches=200,
)
````

## How to test

- Please check the generated docs and see if the two new subpackages appear nicely. 
- Please check whether no func was removed or does not appear in the docs. This is hard to spot from the diff alone since the files were renamed.
- Also can try out the new beamforming pipeline.

## Summary by CodeRabbit

# Release Notes

* **New Features**
  * Added comprehensive signal processing operations framework (`Pipeline`, `Beamform`, etc.)
  * Introduced ultrasound utilities (demodulation, envelope detection, IQ conversion)
  * New tensor operations including `normalize` function

* **Documentation**
  * Updated import paths in examples and tutorials to reflect reorganized API structure

* **Refactor**
  * Reorganized public API into structured modules: `zea.func` and `zea.ops` with submodules for easier discovery
  * Simplified access to common functions through top-level namespace imports

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by tristan-deep** [comment]
Using this snippet now to test the new setup. @vincentvdschaft can you confirm the DMAS output? Looks like bit worse contrast is obtained.

```python
import matplotlib.pyplot as plt
import numpy as np

import zea
from zea.ops import (
    Beamform,
    Demodulate,
    EnvelopeDetect,
    LogCompress,
    Normalize,
)

zea.visualize.set_mpl_style()
path = "hf://zeahub/picmus/database/experiments/contrast_speckle/contrast_speckle_expe_dataset_rf/contrast_speckle_expe_dataset_rf.hdf5"

idx = 30
data, scan, probe = zea.data.load_file(
    path=path,
    indices=(slice(0, 1, None), slice(idx, idx + 7, None)),
    data_type="raw_data",
)

scan.zlims = (1e-3, 40e-3)
scan.xlims = (-20e-3, 20e-3)
num_patches = 256

images = {}
for beamformer_type in ["das", "dmas"]:
    operations = [
        Demodulate(),
        Beamform(beamformer=beamformer_type, pfield=False, num_patches=num_patches),
        EnvelopeDetect(),
        Normalize(),
        LogCompress(clip=False),
    ]

    pipeline = zea.ops.Pipeline(operations=operations, jit_options="ops")
    parameters = pipeline.prepare_parameters(probe, scan)
    parameters["f_number"] = 1.0
    images[beamformer_type] = pipeline(data=data, **parameters)["data"][0]


# Histogram matching: match DMAS to DAS
def histogram_match(source, reference):
    src_sorted = np.sort(source.flatten())
    ref_sorted = np.sort(reference.flatten())
    return np.interp(source.flatten(), src_sorted, ref_sorted).reshape(source.shape)


images["dmas"] = histogram_match(images["dmas"], images["das"])

# Plot side by side
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 5))
extent = [x * 1e3 for x in [*scan.xlims, *scan.zlims]]

for ax, name in zip([ax1, ax2], ["das", "dmas"]):
    ax.imshow(images[name], cmap="gray", extent=extent, origin="lower", vmin=-40, vmax=0)
    ax.set_title(name.upper())
    ax.set_xlabel("Lateral [mm]")
    ax.set_ylabel("Depth [mm]")
    ax.invert_yaxis()

plt.tight_layout()
plt.savefig("test_comparison.png")
```

test_comparison

File: zea/backend/__init__.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ class on_device:
         .. code-block:: python
 
             with zea.backend.on_device("gpu:3"):
-                pipeline = zea.Pipeline([zea.keras_ops.Abs()])
+                pipeline = zea.Pipeline([zea.ops.Abs()])
                 output = pipeline(data=keras.random.normal((10, 10)))  # output is on "cuda:3"
```
[the remark this answers is not in the document]
Comment (tristan-deep) [reply zea/backend/__init__.py]: we accept both gpu and cuda for either backends and they will be correct [here](https://github.com/tue-bmd/zea/blob/859b506e5f22318281900845d98a8fc1aa5aad65/zea/backend/__init__.py#L168).

File: zea/ops/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,1295 @@
...
+                        # This is a single operation branch
+                        branch_op_cls = get_ops(branch_config["name"])
+                        branch_params = branch_config.get("params", {})
+                        branch = branch_op_cls(**branch_params)
+
+                    branches.append(branch)
+
+                # Create the branched pipeline instance
+                operation_instance = operation_cls(branches=branches, **params)
+            # Check for nested operations at the same level as params
```
[the remark this answers is not in the document]
Comment (tristan-deep) [reply zea/ops/pipeline.py]: Not scope for this PR, please create issue for this.

File: zea/ops/pipeline.py [hunk 3]
```diff
@@ -0,0 +1,1295 @@
...
+            branches = []
+
+        self.branches = {}
+        for i, branch in enumerate(branches, start=1):
+            branch_name = f"branch_{i}"
+            # Convert different branch specification types
+            if isinstance(branch, list):
+                # Convert list to operation chain
+                self.branches[branch_name] = make_operation_chain(branch)
+            elif isinstance(branch, (Pipeline, Operation)):
```
[the remark this answers is not in the document]
Comment (tristan-deep) [reply zea/ops/pipeline.py]: Same again, not scope for this PR, please create issue for this (or add to existing branch issue you created).

File: zea/ops/pipeline.py [hunk 4]
```diff
@@ -0,0 +1,1295 @@
...
+            elif isinstance(branch, list):
+                # Convert list of operations to list of operation configs
+                branch_op_configs = []
+                for op in branch:
+                    branch_op_configs.append(op.get_config())
+                branch_configs[branch_name] = {"operations": branch_op_configs}
+            else:
+                # Single operation
+                branch_configs[branch_name] = branch.get_config()
+
```
[the remark this answers is not in the document]
Comment (tristan-deep) [reply zea/ops/pipeline.py]: Please add to #206

**Comment by tristan-deep** [comment]
> I wonder if we even need the `DelayAndSum` (or `DAS`) operation, since it is basically two consecutive `zea.ops.Sum()`'s, which is actually cleaner to me. And we initialize a beamformer using `Beamform(beamformer="das"), it doesn't matter anyway.

I kind of agree, I actually implemented it like that at first. For DAS we can still do that, however for other beamformers it get's more involved. A simple stack of existing base ops might not get you there...

**Review by wesselvannierop — COMMENTED** [review]
API looks good to me, as do the changes. I will test a bit now.

File: tests/test_operations.py [hunk 5]
```diff
@@ -311,8 +317,6 @@ def test_hilbert_transform():
 
     import keras
 
-    from zea import ops
```
Comment (wesselvannierop) [inline tests/test_operations.py]: Interesting, I thought we had to import zea functions inside the `backend_equality_check` for the backend switching to work. But I checked and this seems to work fine now...

**Review by wesselvannierop — DISMISSED** [review]
LGTM! Tested it with the beamforming script I'm working on currently

**Review by wesselvannierop — APPROVED** [review]
